crates/bevy_ecs/src/scheduling/state.rs

```diff
@@ -38,27 +71,33 @@ pub struct OnUpdate<S: States>(pub S);
 /// The current state value can be accessed through this resource. To *change* the state,
 /// queue a transition in the [`NextState<S>`] resource, and it will be applied by the next
 /// [`apply_state_transition::<S>`] system.
-#[derive(Resource)]
+///
+/// The starting state is defined via the [`Default`] implementation for `S`.
+#[derive(Resource, Default)]
 pub struct State<S: States>(pub S);
 
 /// The next state of [`State<S>`].
 ///
 /// To queue a transition, just set the contained value to `Some(next_state)`.
-#[derive(Resource)]
+#[derive(Resource, Default)]
 pub struct NextState<S: States>(pub Option<S>);
 
+impl<S: States> NextState<S> {
+    /// Queue the transition to a new `state`.
+    pub fn queue(&mut self, state: S) {
```

Comment:
I think I like calling this `set`. `queue` feels implied given that the name of this is `NextState`. We are setting the "next state" right now. We aren't queuing the value of `NextState`.